File: DFS&BFS/python/2606.py
n = int(input())
arr = [[] for _ in range(n+1)]
for _ in range(int(input())):
  a,b = map(int,input().split())
  arr[a].append(b)
  arr[b].append(a)
visited = [False for _ in range(n+1)]

def dfs (i):
  visited[i] = True
  for node in arr[i]:
    if (visited[node]) : continue
    visited[node] = True
    dfs(node)

dfs(1)
cnt = -1
for node in visited:
  if node: cnt+=1
print(cnt)

File: DFS&BFS/javascript/2606.js
const input = require("fs").readFileSync("/dev/stdin").toString().split("\n");

const n = Number(input[0]);
const arr = Array.from(Array(n + 1), () => []);
const visited = new Array(n + 1);
for (let i = 2; i < input.length - 1; i++) {
  input[i] = input[i].split(" ");
  arr[Number(input[i][0])].push(Number(input[i][1]));
  arr[Number(input[i][1])].push(Number(input[i][0]));
}
let cnt = -1;
const dfs = (i) => {
  visited[i] = true;
  cnt++;
  arr[i].forEach((node) => {
    if (visited[node]) return;
    visited[node] = true;
    dfs(node);
  });
};

dfs(1);
console.log(cnt);
